components/collapsible-panel/collapsible-panel.js

```diff
@@ -364,6 +382,12 @@ class CollapsiblePanel extends FocusMixin(RtlMixin(LitElement)) {
 		}
 	}
 
+	_handleHeaderStartSlotChange(e) {
```

Comment:
I'm tempted to suggest naming this `_handleHeaderBeforeSlotChange` to be consistent with the "before" slot name. It almost makes it sounds like it's for a fictitious event that is dispatched before the slot change happens. 🤷‍♂️ I'm content either way.